lib/db.php

```diff
@@ -94,17 +94,23 @@ static public function execute($dbh, $sql, $args, $type = false, $all = false) {
         if (!$type) {
             $type = self::execute_type($sql);
         }
-        $sql = $dbh->prepare($sql);
-        if (!$sql || !$sql->execute($args)) {
+        try {
+            $sql = $dbh->prepare($sql);
+            if (!$sql || !$sql->execute($args)) {
+                return false;
+            }
+            if ($type == 'modify' || $type == 'insert') {
+                return $sql->rowCount();
+            }
+            if ($all) {
+                return $sql->fetchAll(PDO::FETCH_ASSOC);
+            }
+            return $sql->fetch(PDO::FETCH_ASSOC);
+        } catch (PDOException $oops) {
+            Hm_Msgs::add('ERRDatabase error. Please try again.');
+            Hm_Debug::add($oops->getMessage());
```

Comment:
if we keep `self::$dbh[$key]->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);` we could then Log the error for later inspection like this: `error_log($e->getMessage());` or use `PDO::ERRMODE_WARNING`

This can be a security issue in production.